core/struct/keyword_init_spec.rb

```diff
@@ -0,0 +1,21 @@
+require_relative '../../spec_helper'
+
+ruby_version_is "3.1" do
+  # See https://bugs.ruby-lang.org/issues/18008
+  describe "StructClass#keyword_init?" do
+    it "returns true for a struct that accepts keyword arguments to initialize" do
```

Comment:
The description is maybe a bit verbose but i borrowed it from `Struct`'s `new_spec.rb` for the sake of consistency. Open to make it shorter if requested.